### PR TITLE
Add a preference to mute all sounds

### LINKED ISSUE #7950: Sound after an unsuccessful search cannot be disabled
### Description of the Issue
After doing an unsuccessful search, the system 'wrong!' error bell is played which I find very annoying but I cannot find a way to disable it. Disabling all sounds or the system error sound in the OS is not a solution because in other cases I do want to hear the error bell, just not after an unsuccessful search  (why would it be an error anyway?)

### Steps to Reproduce the Issue
Search for word that is not in a text file

### Expected Behavior
I would expect to be able to disable the system error bell after an unsuccessful search somewhere in the settings

### Actual Behavior
The system error bell after an unsuccessful search cannot be disabled.

The help file says:
https://npp-user-manual.org/docs/searching/
In addition, the system error bell will sound if a Find Next or Replace action results in the Find what text not being encountered; the bell cannot be disabled.

A similar issue is #2049
where MetaChuh commented on Apr 24, 2019
"please update your notepad++ to the latest version Notepad++ 7.6.6 and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance to you, please feel free to file a new issue as easy to understand and reproducible as possible.
thank you for your help and best regards."

But I have a even newer version and I cannot disable the annoying error bell.

### Debug Information
Notepad++ v7.8.4   (64-bit)
Build time : Jan 29 2020 - 01:31:03
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Home (64-bit) 
OS Version : 1903
OS Build : 18362.657
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll

### PR DESCRIPTION
**Pull request by mere-human**
Fix #7950

### CODE REVIEW

**Comment by mere-human** [comment]
The new preference is located under the auto-updater check-box in the MISC section.
image

It allows to disable various sounds e.g. when a search has no match in a file.

**Comment by sasumner** [comment]
IIRC, @donho was happy with the `PlaySound` implementation as it is, and didn't want an extra preference setting for this.
The sounds not relating to "unsuccessful" search are rare and have never been mentioned by a user AFAIK.

**Comment by mere-human** [comment]
> The sounds not relating to "unsuccessful" search are rare and have never been mentioned by a user AFAIK.

See #7950
They still complain about these sounds. I think it is better to make an easy option to disable it once and for all.

I can make the option elsewhere or rename it or add a scrollbar in the MISC section.

**Comment by mere-human** [comment]
The Notepad++ appears in the volume mixer only after it tries to make a sound.
Disabling the volume this way is not easy and not intuitive.

**Comment by sasumner** [comment]
@mere-human 

Hmm, surprised you didn't "ask to be assigned" to this one before just "fixing" it.  Of course, not an absolute requirement--to be assigned--but it can shortcut potentially wasted time if an issue has a "history".  Hmm, lots of air-quotes on this :-).

Well, you put in some work, let's see if Don has had a change of mind on this.

**Comment by Codingale** [comment]
I believe the Search category isn't a terrible place instead of Misc. I just hate that Windows never mutes it for longer than one restart or even the same session sometimes. On top of that, it's not like having one preference check is going to cause the entire codebase to run slower by a noticeable amount unless the implementation was done poorly. `PlaySound` also  doesn't respect Windows Sound Settings, as I have everything off and it still plays some beep from some archaic method that does what it wants.

**Comment by mere-human** [comment]
> @mere-human
> 
> Hmm, surprised you didn't "ask to be assigned" to this one before just "fixing" it. Of course, not an absolute requirement--to be assigned--but it can shortcut potentially wasted time if an issue has a "history". Hmm, lots of air-quotes on this :-).
> 

Yes, sorry for rushing.
I just had some free time

**Comment by donho** [comment]
@sasumner 
> IIRC, @donho was happy with the PlaySound implementation as it is, and didn't want an extra preference setting for this.
The sounds not relating to "unsuccessful" search are rare and have never been mentioned by a user AFAIK.

I was indeed. However, I've received several users' feedback for this requests because they don't find this option in Notepad++ : disable it in the system sound notification might be a solution but it's not certainly an obvious place for users.

**Review by donho — CHANGES_REQUESTED** [review]
Revert 
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/d3cb2ceb360c8f0d4219929466b2a7330de08597

File: PowerEditor/src/Parameters.h [hunk 1]
```diff
@@ -838,6 +838,7 @@ struct NppGUI final
 	bool _monospacedFontFindDlg = false;
 	bool _findDlgAlwaysVisible = false;
 	bool _confirmReplaceInAllOpenDocs = true;
+	bool _muteSounds = true;
```
Comment (mere-human) [inline PowerEditor/src/Parameters.h:841]: Should I make it "false" by default?

File: PowerEditor/src/WinControls/Preference/preference.rc [hunk 2]
```diff
@@ -416,7 +416,8 @@ BEGIN
     CONTROL         "Peek on document map",IDC_CHECK_ENABLEDOCPEEKONMAP,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,269,71,140,10
 
 	// "Enable Notepad++ auto-updater" should be always the 1st item, because it'll be hidden if GUP.exe is absent
-    CONTROL         "Enable Notepad++ auto-updater",IDC_CHECK_AUTOUPDATE,"Button",BS_AUTOCHECKBOX | BS_MULTILINE | WS_TABSTOP,37,94,217,10
+    CONTROL         "Mute all sounds", IDC_CHECK_MUTE_SOUNDS, "Button", BS_AUTOCHECKBOX | BS_MULTILINE | WS_TABSTOP, 37, 94, 217, 10
```
Comment (mere-human) [inline PowerEditor/src/WinControls/Preference/preference.rc:419]: Should I add a localization for this string?

**Comment by sasumner** [comment]
>So please Revert d3cb2ce

Don't forget to revert the docs (see https://github.com/notepad-plus-plus/npp-usermanual/commit/40dbb47f889de157fe2c0812b079e5bd8f820048 ) as well, when you are adding documentation about the new Preference setting

File: PowerEditor/src/Parameters.h (hunk 1 above)
Comment (donho) [reply PowerEditor/src/Parameters.h:841]: Yes, please.

File: PowerEditor/src/WinControls/Preference/preference.rc (hunk 2 above)
Comment (donho) [reply PowerEditor/src/WinControls/Preference/preference.rc:419]: Yes, please.
As well, add the modification in `english.xml`.

File: PowerEditor/src/Parameters.h (hunk 1 above)
Comment (sasumner) [reply PowerEditor/src/Parameters.h:841]: I think the general idea is that existing behavior should be the default, so that is why this should be `false`

**Comment by mere-human** [comment]
> > So please Revert [d3cb2ce](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/d3cb2ceb360c8f0d4219929466b2a7330de08597)
> 
> Don't forget to revert the docs (see [notepad-plus-plus/npp-usermanual@40dbb47](https://github.com/notepad-plus-plus/npp-usermanual/commit/40dbb47f889de157fe2c0812b079e5bd8f820048) ) as well, when you are adding documentation about the new Preference setting

I will update the manual as soon as this PR is submitted.

**Review by donho — APPROVED** [review]

**Pull request closed without merging**